Shared/Utils/Address.swift

```diff
@@ -95,13 +95,11 @@ struct Address: Hashable, ExpressibleByStringInterpolation, CustomStringConverti
         hexadecimal.prefix(6) + "…" + hexadecimal.suffix(4)
     }
 
-    init(_ value: String, isERC681: Bool) throws {
-        var addressString = value
-        if isERC681 {
-            addressString = Self.addressFromERC681(addressString)
-        }
-
-        _store = try EthereumAddress(hex: addressString, eip55: false)
+    // This will check if ERC681 or EIP3770
+    static func addressWithPrefix(text: String) throws -> (prefix: String?, address: Address) {
```

Comment:
Would it be better to make prefix as a property of `Address`?

Reply:
I am also not sure how do we add prefixes to addresses where we need to present them in the UI. I would assume it to be a part of the Address description if it has a prefix set.

Reply:
as prefix is the short name of corresponding network, is taken from there now